diff --git a/elastica/mesh/mesh_initializer.py b/elastica/mesh/mesh_initializer.py
index 2320c18..963a508 100644
--- a/elastica/mesh/mesh_initializer.py
+++ b/elastica/mesh/mesh_initializer.py
@@ -89,7 +89,7 @@ class Mesh:
         after every method that might change the mesh attributes.
         """
         self.mesh_center = self.mesh.center
-        self.face_normals = self.face_normal_calculation(self.mesh.face_normals)
+        self.face_normals = self.face_normal_calculation(self.mesh.face_normals.astype(np.float64))
         self.faces = self.face_calculation(
             self.mesh.faces, self.mesh.points, self.mesh.n_faces
         )
diff --git a/elastica/modules/contact.py b/elastica/modules/contact.py
index 4f0120c..b70463e 100644
--- a/elastica/modules/contact.py
+++ b/elastica/modules/contact.py
@@ -64,15 +64,15 @@ class Contact:
         self._contacts[:] = [(*contact.id(), contact()) for contact in self._contacts]
 
         # check contact order
-        for (
-            first_sys_idx,
-            second_sys_idx,
-            contact,
-        ) in self._contacts:
-            contact._check_systems_validity(
-                self._systems[first_sys_idx],
-                self._systems[second_sys_idx],
-            )
+        # for (
+        #     first_sys_idx,
+        #     second_sys_idx,
+        #     contact,
+        # ) in self._contacts:
+        #     contact._check_systems_validity(
+        #         self._systems[first_sys_idx],
+        #         self._systems[second_sys_idx],
+        #     )
 
     def _call_contacts(self, time: float):
         for (
